core/services/ocr2/plugins/ocr2keeper/evm21/registry.go

```diff
@@ -926,3 +937,76 @@ func (r *EvmRegistry) getBlockHash(blockNumber *big.Int) (common.Hash, error) {
 
 	return block.Hash(), nil
 }
+
+func (r *EvmRegistry) getTxBlock(txHash common.Hash) (*big.Int, common.Hash, error) {
+	txr, err := r.client.TransactionReceipt(r.ctx, txHash)
+	if err != nil {
+		return nil, common.Hash{}, err
+	}
+
+	return txr.BlockNumber, txr.BlockHash, nil
+}
+
+func (r *EvmRegistry) getCheckResult(p ocr2keepers.UpkeepPayload, reason, state uint8) ocr2keepers.CheckResult {
+	return ocr2keepers.CheckResult{
+		IneligibilityReason:    reason,
+		PipelineExecutionState: state,
+		UpkeepID:               p.UpkeepID,
+		Trigger:                p.Trigger,
+		WorkID:                 p.WorkID,
+		FastGasWei:             big.NewInt(0),
+		LinkNative:             big.NewInt(0),
+	}
+}
+
+// verifyCheckBlock checks that the check block and hash are valid
+func (r *EvmRegistry) verifyCheckBlock(checkBlock, upkeepId *big.Int, checkHash common.Hash, p ocr2keepers.UpkeepPayload, i int, results []ocr2keepers.CheckResult) bool {
```

Comment:
return error and set the result from the caller

Reply:
done